### DISCUSSION TITLE
Dependencies accidentally added to the path operation function instead of the path operation decorator are silently ignored

### DISCUSSION THREAD

**Discussion by mattalbr** [fastapi/fastapi#8458] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Depends, FastAPI, Header, HTTPException

app = FastAPI()

async def verify_token(x_token: str = Header(...)):
    if x_token != "fake-super-secret-token":
        raise HTTPException(status_code=401, detail="X-Token header invalid")

@app.get("/")
def hello_world(dependences=[Depends(enforce_login)]) -> str:
    return "Hello World!"
```

### Description

* Open the browser and call endpoint "/"
* It returns "Hello World!"
* It should issue a warning, fail to deploy, or return a 401 code.

Opening this for discussion purposes. I just did this myself so it's not just theoretical. No clue if I'm the first or only to have made this mistake, but seems unlikely.

The idea behind dependencies in decorators is to make use of dependencies whose return value we can ignore. One super obvious use case, that's actually mentioned in the docs, is verifying some access token. The problem is, not running the verification looks the same as passing verification, and it's really easy (I just did it) to accidentally toss the "dependencies=[]" on the function instead of the decorator. What happens then is you think your function is checking auth, but it's not, which is Bad (TM).

I'm not sure exactly what the "correct" behavior should be, this just seems like a classic case of needles next to poison-tipped needles that could easily slip by code review. Thoughts? I will note that a good linter catches the mutable object as default value.

### Operating System

macOS

### FastAPI Version

0.75.0

### Python Version

3.10.2

**Comment by jonatasoli** [comment]
In this case your propose check is function/class exist in Depends statement? I believe that in this case the issue needs to change its label.

**Comment by mattalbr** [comment]
The issue isn't so much the function/class not existing in the Depends statement as it is the fact that the Depends is never getting resolved because it's being put into an unused kwarg underneath a list. Unfortunately it's probably too heavyweight, but it would almost be nice to have a warning in the destructor of Depends if it was never resolved. 

It's totally possible that there's no good solution to this problem, but I thought it was worth bringing up since it bit me.
